improver/uv_index.py

```diff
@@ -97,12 +97,13 @@ def calculate_uv_index(
         )
         raise ValueError(msg)
 
+    uv_downward.convert_units("W m-2")
     uv_data = uv_downward.data * scale_factor
     attributes = generate_mandatory_attributes(
         [uv_downward], model_id_attr=model_id_attr
     )
     uv_index = create_new_diagnostic_cube(
-        "ultraviolet_index", "1", uv_downward, attributes, data=uv_data
+        "ultraviolet_index", "W m-2", uv_downward, attributes, data=uv_data
```

Comment:
No. The units for the UV index were correctly unitless. The point of the scale factor is that a UV index of 10 relates to a UV flux of 10/3.6 W m-2.